MagneticField/VolumeBasedEngine/src/MagGeometry.cc

```diff
@@ -20,6 +20,12 @@
 using namespace std;
 using namespace edm;
 
+namespace {
+  std::atomic<int> instanceCounter(0);
```

Comment:
@namapane Could you add a comment along
```suggestion
  // A thread-local cache is accepted because MagneticField is used by several other ESProducts
  // via member variable, MagneticField and the other ESProducts are widely used, and migrating
  // all the uses of all those was deemed to have very high cost.
  std::atomic<int> instanceCounter(0);
```
to document to rationale?